### PR TITLE
Fix Vale violations across the Kedro documentation

### PR DESCRIPTION
**Pull request by deepyaman**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by stichbury** [comment]
I know this isn't draft and you've not yet asked for my feedback, but just to say that we could drive ourselves mad if we take all the vale feedback on in one go, and I'm not sure it's super productive. Much of the advanced docs need solid rewrite for tech changes, usability and to extend them to add extra examples, so they will change anyway. And although Vale fixes may improve cosmetically it isn't to say that the docs are perfect because they're passing. We don't want to get to the point where we have a set of perfectly phrased but totally unhelpful docs because they're plain out of date or whatever.

**Tl;dr**: I'm happy to accept that there are infractions to Vale's rules within our docs. We'll and fix things as we get there. 

That said, I really appreciate you adding the changes! Just I wouldn't commit too much of your time in this direction at present, since it's not a perfect fix.

**Comment by deepyaman** [comment]
> I know this isn't draft and you've not yet asked for my feedback, but just to say that we could drive ourselves mad if we take all the vale feedback on in one go, and I'm not sure it's super productive. Much of the advanced docs need solid rewrite for tech changes, usability and to extend them to add extra examples, so they will change anyway. And although Vale fixes may improve cosmetically it isn't to say that the docs are perfect because they're passing. We don't want to get to the point where we have a set of perfectly phrased but totally unhelpful docs because they're plain out of date or whatever.
> 
> **Tl;dr**: I'm happy to accept that there are infractions to Vale's rules within our docs. We'll and fix things as we get there.
> 
> That said, I really appreciate you adding the changes! Just I wouldn't commit too much of your time in this direction at present, since it's not a perfect fix.

All makes sense! There are a ton of Vale violations, and a lot of them aren't even correct. However, some small percentage of them are (e.g. misspellings, duplicated words, mismatched case, American spelling), so I'm just taking a pass through to catch the ones that seem like more clear improvements.

**Comment by stichbury** [comment]
> All makes sense! There are a ton of Vale violations, and a lot of them aren't even correct. However, some small percentage of them are (e.g. misspellings, duplicated words, mismatched case, American spelling), so I'm just taking a pass through to catch the ones that seem like more clear improvements.

That's great then, thanks! Definitely got to eliminate that American spelling 😛

**Comment by deepyaman** [comment]
@stichbury This is ready to go! I leave it up to your judgement if want to revert any of the changes I've made, or don't like them, but I'd say 75% of the changes are clear improvements and the other 25% sounded like improvements to me (but, e.g., if like the use of "and/or" in a lot of places, feel free to revert).

**Review by astrojuanlu — COMMENTED** [review]
Gave a quick look, approving if @stichbury approves - thanks @deepyaman!

File: docs/source/configuration/configuration_basics.md
Comment (stichbury) [inline docs/source/configuration/configuration_basics.md:201]:
```suggestion
Customise the configuration loader arguments in `settings.py` as follows if your project does not have any other environments apart from `base` (i.e. no `local` environment to default to):
```

File: docs/source/configuration/credentials.md
Comment (stichbury) [inline docs/source/configuration/credentials.md:52]:
```suggestion
When you work with AWS credentials on datasets, you are not required to store AWS credentials in the project configuration files. Instead, you can specify them using environment variables `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and, optionally, `AWS_SESSION_TOKEN`. See the [official AWS documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) for more details.
```

File: docs/source/data/data_catalog.md
Comment (stichbury) [inline docs/source/data/data_catalog.md:35]:
```suggestion
Kedro supports a range of connectors, for CSV files, Excel spreadsheets, Parquet files, Feather files, HDF5 files, JSON documents, pickled objects, SQL tables, SQL queries, and more. They are supported using libraries such as pandas, PySpark, NetworkX, and Matplotlib.
```

File: docs/source/data/data_catalog.md
Comment (stichbury) [inline docs/source/data/data_catalog.md:49]:
```suggestion
- **S3 Compatible Storage**: `s3://my-bucket-name/path/_to/data` - for example, MinIO, using the s3fs library.
```

File: docs/source/data/data_catalog_yaml_examples.md
Comment (stichbury) [inline docs/source/data/data_catalog_yaml_examples.md:372]:
```suggestion
Define two `DataCatalog` entries for the same dataset in a common format (for example, Parquet, JSON, CSV) in your `conf/base/catalog.yml`:
```

File: docs/source/data/how_to_create_a_custom_dataset.md
Comment (stichbury) [inline docs/source/data/how_to_create_a_custom_dataset.md:514]:
```suggestion
There is one dataset that is an exception: [SparkDataSet](/kedro_datasets.spark.SparkDataSet). [Apache Spark](https://spark.apache.org/) uses its own parallelism and therefore doesn't work with [ParallelRunner](/kedro.runner.ParallelRunner). For parallelism within a Kedro project that uses Spark, use [ThreadRunner](/kedro.runner.ThreadRunner) instead.
```

File: docs/source/data/how_to_create_a_custom_dataset.md
Comment (stichbury) [inline docs/source/data/how_to_create_a_custom_dataset.md:572]:
```suggestion
We provide additional examples of [how to use parameters through the data catalog's YAML API](./data_catalog_yaml_examples.md). For an example of how to use these parameters in your dataset constructor, see the [SparkDataSet](/kedro_datasets.spark.SparkDataSet) implementation.
```

**Review by stichbury — APPROVED** [review]
This is great, thanks @deepyaman. I made a couple of suggestions (mostly just to remove "please" as it's not necessary, even if it is polite!) and i'd prefer there to be no `i.e.` and `e.g.` but didn't get too hung up on it. Whatever changes you make will be better so let's go for it! Approved 🏆 🙏

File: docs/source/data/data_catalog.md
Comment (deepyaman) [reply docs/source/data/data_catalog.md:35]: @stichbury For my knowledge, should there be an Oxford comma here? Wondering because I came across a lot of instances where Vale wanted to add an Oxford comma, but I assumed not using one was the British way.

File: docs/source/deployment/amazon_emr_serverless.md
Comment (deepyaman) [inline docs/source/deployment/amazon_emr_serverless.md:97]:
```suggestion
- [Customizing an EMR Serverless image](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/application-custom-image.html)
```
Didn't mean to change this; the title has American spelling in the target article.

File: docs/source/data/data_catalog.md
Comment (stichbury) [reply docs/source/data/data_catalog.md]: I've been adopting the Oxford comma as Vale requests it. I am not a big fan of it but it does have value in tech writing for clarity so where it's missing, and flagged as absent, it is fine to add it in.

File: docs/source/data/data_catalog.md
Comment (deepyaman) [inline docs/source/data/data_catalog.md:35]:
```suggestion
Kedro supports a range of connectors, for CSV files, Excel spreadsheets, Parquet files, Feather files, HDF5 files, JSON documents, pickled objects, SQL tables, SQL queries and more. They are supported using libraries such as pandas, PySpark, NetworkX and Matplotlib.
```
Remove Oxford commas?

Comment (stichbury) [reply docs/source/data/data_catalog.md:35]: Let's leave them! I am (relatively) happy to use them for clarity and as Vale spots them missing throughout the docs in weeks to come, I'll add them in rather than take them out.

**Review by astrojuanlu — APPROVED** [review]

**Pull request merged**
